packages/berry-pnpify/sources/generateSdk.ts

```diff
@@ -2,47 +2,73 @@ import {Filename, NodeFS, PortablePath, xfs, ppath} from '@berry/fslib';
 
 import {dynamicRequire}                             from './dynamicRequire';
 
-const TEMPLATE = (relPnpApiPath: string) => [
+const TEMPLATE = (relPnpApiPath: string, module: string, pnpify: boolean) => [
   `const relPnpApiPath = ${JSON.stringify(NodeFS.toPortablePath(relPnpApiPath))};\n`,
   `const absPnpApiPath = require(\`path\`).resolve(__dirname, relPnpApiPath);\n`,
   `\n`,
-  `// Setup the environment to be able to require @berry/pnpify\n`,
+  `// Setup the environment to be able to require ${module}\n`,
   `require(absPnpApiPath).setup();\n`,
   `\n`,
   `// Prepare the environment (to be ready in case of child_process.spawn etc)\n`,
   `process.env.NODE_OPTIONS = process.env.NODE_OPTIONS || \`\`;\n`,
   `process.env.NODE_OPTIONS += \` -r \${absPnpApiPath}\`;\n`,
-  `process.env.NODE_OPTIONS += \` -r \${require.resolve(\`@berry/pnpify/lib\`)}\`;\n`,
-  `\n`,
-  `// Apply PnPify to the current process\n`,
-  `require(\`@berry/pnpify/lib\`).patchFs();\n`,
-  `\n`,
-  `// Defer to the real typescript your application uses\n`,
-  `require(\`typescript/lib/tsserver\`);\n`,
+  ...(pnpify ? [
+    `process.env.NODE_OPTIONS += \` -r \${require.resolve(\`@berry/pnpify/lib\`)}\`;\n`,
+    `\n`,
+    `// Apply PnPify to the current process\n`,
+    `require(\`@berry/pnpify/lib\`).patchFs();\n`,
+    `\n`,
+  ] : []),
+  `// Defer to the real ${module} your application uses\n`,
+  `module.exports = require(\`${module}\`);\n`,
 ].join(``);
 
-export async function generateSdk(projectRoot: PortablePath, targetFolder: PortablePath | null) {
-  if (targetFolder === null)
-    targetFolder = projectRoot;
+const addVSCodeWorkspaceSettings = async (projectRoot: PortablePath, settings: any) => {
+  const settingsPath = ppath.join(projectRoot, `.vscode/settings.json` as PortablePath);
+  const content = await xfs.existsPromise(settingsPath) ? await xfs.readFilePromise(settingsPath, `utf8`) : `{}`;
 
-  const tssdk = ppath.join(targetFolder, `tssdk` as PortablePath);
-  const tssdkManifest = ppath.join(tssdk, `package.json` as PortablePath);
-  const tsserver = ppath.join(tssdk, `lib/tsserver.js` as PortablePath);
+  const data = JSON.parse(content);
+  const patched = `${JSON.stringify({...data, ...settings}, null, 2)}\n`;
+
+  await xfs.mkdirpPromise(ppath.dirname(settingsPath));
+  await xfs.changeFilePromise(settingsPath, patched);
+}
+
+const generateTypescriptWrapper = async (projectRoot: PortablePath, tssdk: PortablePath) => {
+  const typescript = ppath.join(tssdk, `typescript` as PortablePath);
```

Comment:
Since at the moment `tssdk` doesn't point to the sdk anymore, I'd suggest to rename it to `target`. This way we can continue to use the `tssdk` name to refer to the actual `typescript.tssdk` setting.

```suggestion
  const tssdk = ppath.join(target, `typescript` as PortablePath);
```